apache/metron
# METRON-1702: Reload a running job in the UI

## Contributor Comments
This PR update the Pcap query panel so that a currently running query for a user is loaded when a page is refreshed.

### Changes Included

- Updated pcap-panel.component.ts to query for a running job, and if found, update the current status and start polling
- Refactored pcap-panel.component.ts by moving updating status and start polling into separate functions.  This should make the class more readable and is needed to update status after querying for a running job.
- Updated pcap-filters.component.ts to accept a PcapRequest as input and update filters to match the request made for the currently running job
- Added functions to pcap.service.ts to get the currently running job and the pcap request for that job
-  Fixed a bug in PcapServiceImpl.java where the wrong types were being used for some fields (also adjust unit tests to catch these errors in the future)
- Added tests for new code

### Testing
This has been tested in full dev. 

1. Navigate to the Pcap panel in the Alerts UI
2. Submit a query with filters and wait for progress to be > 0
3. Refresh the page
4. The page should look exactly the same (filters and all) and polling should continue until the job is complete

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
